File: python/0242有效的字母异位词.py


def isAnagram1(s: str, t: str) -> bool:
    memo=dict()

    for i in s:
        if(i not in memo.keys()):
            memo[i]=1
        else:
            memo[i]+=1
        
    for i in t:
        if(i not in memo.keys()):
            return False
        
        memo[i]-=1
        if(memo[i]<0):
            return False

    return False if sum(memo.values()) else True


def isAnagram2(s: str, t: str) -> bool:
    memo=[0]*26

    for i in s:
        memo[ord(i)-97]+=1
    

    for i in t:
        idx=ord(i)-97
        memo[idx]-=1

        if(memo[idx]<0):
            return False

    return False if(sum(memo)) else True

print(isAnagram2("ab","a"))
        

File: cpp/0242有效的字母异位词.cpp
#include<string>

using namespace std;

bool isAnagram(string s, string t)
{
    int memo[26] = {0};

    for(char i : s)
        ++memo[i - 97];
    for(char i : t)
        --memo[i - 97];

    for (int i = 0;i<26;++i)
    {
        if(memo[i])
            return false;
    }

    return true;
}

int main()
{
    
}